features/gateway_injection.md

```diff
@@ -0,0 +1,146 @@
+
+- [x] [Feedback plan](insert_your_link_here). N/A, we already have feedback from users that this is desired and useful.
+
+- [x] Disabled by default.
+
+- [x] No impact on performance when the feature is disabled.
+
+---
+
+## Alpha
+
+### Requirements:
+
+**Design**
+
+- [x] RFC has been approved describing the intention of the feature as well as the user stories behind the feature.
+
+**Config**
+
+- [x] Explicit user action is required to enable this feature (e.g. a config field, config resource, or installation action).
+
+> [Documentation](https://preliminary.istio.io/latest/docs/setup/additional-setup/sidecar-injection/#custom-templates-experimental)
+
+**Docs**
+
+- [x] Reference docs are published to preliminary.istio.io or the Istio wiki.
+  
+There are no reference docs required for this feature.
+
+- [x] Basic feature docs are published on preliminary.istio.io describing what the feature does, how to use it, and any caveats.
+
+> [Documentation](https://preliminary.istio.io/latest/docs/setup/additional-setup/sidecar-injection/#custom-templates-experimental).
+
+- [x] Release notes entries added as appropriate
+- [x] Upgrade notes entries added as appropriate
+
+**Tests**
+
+- [x] Automated integration tests cover core use cases with the feature enabled. [Link](https://github.com/istio/istio/blob/73e56211605aaa80de48f9147c94a05701169995/tests/integration/pilot/ingress_test.go#L406)
+- [x] When disabled, the feature does not affect system stability or performance. The feature does nothing when not enabled.
+
+**API**
+
+- [x] Initial API review. API approved by environments working group
+
+**Approvals**
+
+- [ ] The appropriate work group(s) have reviewed and approved promotion of the feature.
+- [ ] The TOC has reviewed and approved promotion of the feature as part of the
+  roadmap for a release.
+
+---
+
+## Beta
+
+### Requirements:
+
+**Design**
+
+- [x] Design doc describing the intention of the feature, how it will be
+  implemented, and any thoughts on how to test the feature has been approved by
+  relevant work group leads. [Design Doc](https://docs.google.com/document/d/1Rmp9B3DDypgMCau-YAqidx_r3qvKLZj6jUX-t22zj84/).
+- [ ] Feature coverage and test plans written and approved.
+
+**Docs**
+
+- [ ] Documentation on istio.io includes performance expectations; may have caveats.
+- [x] Documentation on istio.io includes samples/tutorials. [Here](https://preliminary.istio.io/latest/docs/setup/additional-setup/sidecar-injection/#custom-templates-experimental) and [here](https://preliminary.istio.io/latest/docs/setup/additional-setup/external-controlplane/) for enabling and using the feature.
+- [x] Documentation on istio.io includes appropriate glossary entries. No glossary entries needed, this uses existing concepts.
```

Comment:
There are no glossary entries for gateway or injection, I think we should probably add some simple ones for those. So I would un-check-mark this since the existing concepts don't have entries either :)

Reply:
Wow, very surprised - I assumed they did. Opened https://github.com/istio/istio.io/pull/9733